packages/controllers/src/permissions/PermissionController.ts

```diff
@@ -0,0 +1,1884 @@
+import {
+  Json,
+  BaseControllerV2 as BaseController,
+  RestrictedControllerMessenger,
+  AddApprovalRequest,
+  AcceptRequest as AcceptApprovalRequest,
+  RejectRequest as RejectApprovalRequest,
+  HasApprovalRequest,
+  StateMetadata,
+} from '@metamask/controllers';
+// These are used in docstrings, but ESLint is ignorant of docstrings.
```

Comment:
Huh. I swear this used to work... we use it all over the place in the extension.

I was missing a single quote in that example (I've just updated it), but even after fixing it, I can't get it to work either. VSCode clearly understands the import because the correct tooltip shows up on hover, but the name "ApprovalController" doesn't get the type assigned to it.

Reply:
It definitely does work in plain JavaScript, but I guess it doesn't work for TypeScript.